src/t_zset.c

```diff
@@ -2698,10 +2700,13 @@ void zunionInterDiffGenericCommand(client *c, robj *dstkey, int numkeysIndex, in
 
                 /* Only continue when present in every input. */
                 if (j == setnum) {
-                    tmp = zuiNewSdsFromValue(&zval);
-                    znode = zslInsert(dstzset->zsl,score,tmp);
-                    dictAdd(dstzset->dict,tmp,&znode->score);
-                    if (sdslen(tmp) > maxelelen) maxelelen = sdslen(tmp);
+                    if (!cardinality_only) {
+                      tmp = zuiNewSdsFromValue(&zval);
```

Comment:
nit: this spacing is off

Reply:
Agreed. Fixed.

Reply:
Still missing the blockage of irrelevant options (`WITHSCORES`, `WEIGHTS`, ...)